src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.NormalSearch.cs

```diff
@@ -55,12 +60,82 @@ public static async Task SearchDocumentInCurrentProcessAsync(
         var (patternName, patternContainerOpt) = PatternMatcher.GetNameAndContainer(searchPattern);
         var declaredSymbolInfoKindsSet = new DeclaredSymbolInfoKindSet(kinds);
 
-        var results = new ConcurrentSet<RoslynNavigateToItem>();
-        await SearchSingleDocumentAsync(
-            document, patternName, patternContainerOpt, declaredSymbolInfoKindsSet, t => results.Add(t), cancellationToken).ConfigureAwait(false);
+        // In parallel, search both the document requested, and any relevant 'related documents' we find for it. For the
+        // original document, search the entirety of it.  For related documents, only search the spans of the
+        // partial-types/inheriting-types that we find for the types in this starting document.
+        var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+
+        await Task.WhenAll(
+            SearchDocumentsInCurrentProcessAsync([(document, new NormalizedTextSpanCollection(new TextSpan(0, text.Length)))]),
```

Comment:
>new NormalizedTextSpanCollection(new TextSpan(0, text.Length) [](http://example.com/codeflow?start=61&length=61)

could just pass in null to indicate that limiting the search span isn't necessary (would remove the need for the GetTextAsync call)

Reply:
sure.  i can do that.

Reply:
done.